core/src/test/kotlin/generator/TestRunner.kt

```diff
@@ -0,0 +1,101 @@
+package generator
+
+import org.jetbrains.dokka.*
+import org.jetbrains.dokka.utilities.DokkaConsoleLogger
+import org.jetbrains.dokka.utilities.DokkaLogger
+import java.io.File
+import java.nio.charset.Charset
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+
+object TestRunner {
+    private var testDir: Path? = null
+
+    fun generateTestDokka(
+        name: String,
+        query: String,
+        checks: Map<DokkaTestGenerator.Companion.Stage, (DokkaTestGenerator.State) -> Boolean>,
+        targets: List<String> = listOf("jvm"),
+        platforms: List<String> = listOf("jvm"),
+        logger: DokkaLogger = DokkaConsoleLogger
+    ) {
+        if (testDir == null) {
+            testDir = Files.createTempDirectory("tests")
+            testDir!!.toFile().deleteOnExit()
+        }
+
+        val root = testDir!!.resolve(name)
+        root.deleteContents()
+
+        val output = root.resolve("out")
+        Files.createDirectories(output)
+
+        platforms.mapNotNull { it.toPlatform() }
+            .forEach {
+                output.deleteContents()
+
+                val config = DokkaConfigurationImpl(
+                    outputDir = output.toString(),
+                    format = "html",
+                    generateIndexPages = true,
+                    passesConfigurations = listOf(
+                        PassConfigurationImpl(
+                            moduleName = name,
+                            sourceRoots = listOf(root).map { SourceRootImpl(it.toString()) },
+                            analysisPlatform = it,
+                            targets = targets
+                        )
+                    )
+                )
+
+                query.toFileMap().materializeFiles(root)
+
+                DokkaTestGenerator(config, logger, checks).generate()
+            }
+    }
+
+    fun String.toFileMap(): Map<String, String> {
+        return if (this.trim().isNotEmpty()) {
+            this.replace("\r\n", "\n")
+                .split("\n/")
+                .map{
+                    it.split("\n".toRegex(), 2)
+                        .let{
+                            it.first().trim().removePrefix("/") to it.last().trim()
+                        }
+                }
+                .toMap()
+        } else {
+            mapOf()
+        }
+    }
```

Comment:
I'm almost sure you can still minify that by using `split("\n").chunked(2) { (k, v) -> k.trim() to v.trim() }.toMap()`

Reply:
Futhermore, you can always replace 
```
fun x() {
    return y
} 
```
with 
``` 
fun x() = y 
```
here and in any other place

Reply:
Basically you're just grouping the paths here in sets of two

Reply:
String.split(r: Regex, n: Int) splits it to n parts, last one may contain r occurrences. In this case it splits it to first line -> rest.

Reply:
`split("\n", limit = 2).let { mapOf(it.first().trim to it.last().trim()) }`